### ISSUE TITLE
Addressing the desertion meta in 4 player games 

### ISSUE THREAD

**Issue by harikmenon** [HaliteChallenge/Halite-II#246]
Here is a replay showcasing what I believe is a core issue to the game design in 4 player games: https://halite.io/play/?game_id=2385549&replay_class=1&replay_name=replay-20171107-215106%2B0000--2378243263-360-240-151009140417

I think most would agree ewirkerman should be ranked in second place, but he ends up ranked last because the other 2 players were having ships playing hide-and-seek. Maybe some would defend this as a valid strategy, but I don't share this opinion, as it muddies the significance of rank 2-4 outcome.

After some discussion on this topic in Discord with other players, here are a few suggestions on how to fix this (only for 4 player games):

Loss triggered after not controlling any planet for X turns. (proposed X = 40)
Loss triggered after not controlling any planet if the game is past Y turns. (proposed Y = 75)
Rank 2 to 4 is determined by total number of ships produced.
Of course please share if you have more opinions. Thanks!

**Comment by succcubbus** [comment]
In case rank 2 to 4 would be determined by total number of ships produced, should timeouts be treated special?
Example: https://halite.io/play/?game_id=2409335

**Comment by succcubbus** [comment]
Also, the proposal by 0x726b00 on the forum would circumvent that:

"[...], at the end of the game rank all players by number of remaining fully docked ships and keep the old tie breakers: 1. total produced ships, 2. total damage dealt. For surviving players this is a move from total past production to current production, which I think more closely resembles the position at the end of the game."

**Comment by lidavidm** [comment]
Hmm. I'm hesitant to drastically rework the ranking criteria, but the loss trigger conditions feel like band-aids to me, so I'm willing to entertain a new set of criteria. However, I do feel that if you are eliminated, no matter how much production you had, you still didn't manage to survive (of course, you could simply have been targeted by another bot while a less-productive rival was passed over, but such is war). Also, as @succcubbus points out, it raises the question of how to handle bots that time out. @j-clap what do you think?

**Comment by mellendo** [comment]
The current rules encourage everyone to program running away when they are nearly dead.  Once everyone programs that, that programming becomes useless, as when winner takes over all planets, the game will end, & the tie breaker (most ships produced?) will be enforced anyways for the multiple survivors.

Rather than force everyone to do this (silly?) code, I think it is better to just apply the tiebreaker to everyone, regardless of who survived.

If you really want to encourage survival, then perhaps change it so the game ends only when 1 bot remains, or 2 bots remain and 1 bot controls all planets.  Then it would be a real fight for survival.  I personally think this would not be an interesting addition to the game.

**Comment by lidavidm** [comment]
One of the victory conditions is to control every planet; if you fully dock a ship to every planet, the game ends there, regardless of the number of surviving bots.

**Comment by mellendo** [comment]
Exactly, that is why the survival programming is silly, because when everyone programs it, it becomes useless, as once the winner docks to every planet, the 3 players that have ships flying aimlessly around are ranked according to the tiebreaker.

**Comment by mellendo** [comment]
So it is better to always apply the tiebreaker, and spare the bots the busy work of programming survival that will become useless once everyone does it.

**Comment by salcan** [comment]
Interesting thread. Any developments since 11/10?
